make_changelog.py

```diff
@@ -0,0 +1,76 @@
+import datetime
+import re
+import sys
+
+# Read existing changelog
+with open("Changelog.md") as f:
+    changelog_lines = list(s.rstrip() for s in f)
+
+# Determine new version
+this_version = sys.argv[1]
+today = datetime.date.today()
+today_str = f"{today.strftime('%Y')}-{today.strftime('%m')}-{today.strftime('%d')}"
+
+# Determine last version
+matches = re.finditer(r"^## \[(.*)\].*$", "\n".join(changelog_lines), re.MULTILINE)
+last_version = next(matches)[1]
+
+# Stop the script if new version is already the latest
+if last_version == this_version:
+    print(f"Version {this_version} is already up-to-date.")
+    sys.exit(0)
+
+# Find line with "## Unreleased" heading
+unreleased_idx = next(
+    i for i, line in enumerate(changelog_lines) if line.startswith("## Unreleased")
+)
+
+# Find line with the last release (i.e. "## x.y.z" heading)
+last_release_idx = next(
+    i
+    for i, line in enumerate(changelog_lines)
+    if line.startswith("## ") and i > unreleased_idx
+)
+
+# Clean up unreleased notes (i.e. remove empty sections)
+unreleased_notes = changelog_lines[(unreleased_idx + 2) : last_release_idx]
+
+empty_unreleased_sections = [
+    (i, line)
+    for i, line in enumerate(unreleased_notes)
+    if line.startswith("### ") and unreleased_notes[i + 1] == ""
```

Comment:
I think, there are two problems with this: 1) there might be a simple empty line after the heading as a "spacer" after which a entry follows <-- in that case, the section would be recognized as unreleased. 2) There might be an empty "bullet point" line which starts with a `-` <-- that section would be recognized as not-empty.

Both cases appear in the current changelog. One example:

```
### Changed

- `BoundingBox` and `Mag` are now immutable attr classes containing `Vec3Int` values. See breaking changes above.

### Fixed

-
```

I'd suggest to split by `### ` and then count the matches for `^- [^\n]+` (with `gm` for regex options)

Reply:
Fixed in https://github.com/scalableminds/webknossos-libs/pull/503/commits/f1e67524c69afc61a7b8c5faa4f6f3e44b032aa6